src/pykeen/utils.py

```diff
@@ -1095,3 +1100,118 @@ def load_configuration(path: Union[str, pathlib.Path, os.PathLike]) -> Mapping[s
+    :returns: Element-wise distance function scores as per the definition above
+
+    points: p
+    box_lows: l
+    box_highs: h
+
+    w = h - l . Width is the difference between the upper and lower box bound
+    c = (h + l) / 2. Box centers (the mean of the box bounds)
+
+    .. math::
+
+        dist(p,l,h) = \begin{cases}
+            |p-c|/(w+1) & l <= p <+ h \\
+            |p-c|*(w+1) - 0.5*w*((w+1)-1/(w+1)) & otherwise \\
+        \end{cases}
+    """
+    widths = box_highs - box_lows
+    widths_p1 = widths + 1  # Compute width plus 1
+    centres = 0.5 * (box_lows + box_highs)  # Compute box midpoints
+    rv = torch.where(
+        torch.logical_and(points >= box_lows, points <= box_highs),
+        torch.abs(points - centres) / widths_p1,  # If true (inside the box)
+        widths_p1 * torch.abs(points - centres) - (0.5 * widths) * (widths_p1 - 1 / widths_p1),
+    )
+    return rv
+
+
+def boxe_kg_arity_position_computation(
+    entity_pos: torch.FloatTensor,
+    other_entity_bump: torch.FloatTensor,
+    relation_box_low: torch.FloatTensor,
+    relation_box_high: torch.FloatTensor,
+    tanh_map: bool,
+    norm_order: int,
+) -> torch.FloatTensor:
+    r"""Perform the BoxE computation at a single arity position.
+
+    :param entity_pos:
+        This is the base entity position of the entity appearing in the target position. For example,
+        for a fact $r(h,t)$ and the head arity position, entity_pos is the base position of $h$.
+    :param other_entity_bump:
+        This is the bump of the entity at the other position in the fact. For example, given a
+        fact $r(h,t)$ and the head arity position, other_entity_bump is the bump of $t$.
+    :param relation_box_low:
+        The lower corner of the relation box at the target arity position.
+    :param relation_box_high:
+        The upper corner of the relation box at the target arity position.
+    :param tanh_map:
+        A Boolean value specifying whether to apply the tanh map regularizer.
+    :param norm_order:
+        The norm order to apply across dimensions to compute overall position score.
+    :return:
+        Arity-position score for the entity relative to the target relation box.
+
+    .. note:: this computation is parallelizable across all positions
+    """
+    bumped_representation = entity_pos + other_entity_bump  # Step 1: Apply the other entity bump
+    if tanh_map:
+        relation_box_low = torch.tanh(relation_box_low)  # Step 2: Apply tanh if tanh_map is set to True.
+        relation_box_high = torch.tanh(relation_box_high)
+        bumped_representation = torch.tanh(bumped_representation)
+    # Compute the distance function output element-wise
+    element_wise_distance = point_to_box_distance(
+        points=bumped_representation, box_lows=relation_box_low, box_highs=relation_box_high
+    )
+    # Finally, compute the norm
+    overall_score = element_wise_distance.norm(p=norm_order, dim=-1)
```

Comment:
for translational models (e.g. TransE), we also support power norm, i..e, `|x|_p^p`, which has some nice numeric properties (and was used by some other interaction models, too). Can we also support this here?

Reply:
![](https://c.tenor.com/7EttrztgjiAAAAAM/palpatine-star-wars.gif)

Reply:
I haven't experimented with this as part of BoxE, but I don't see a reason why this wouldn't be supported for aggregating distance scores.

Reply:
@mberr can take care of this one directly

Reply:
ab6deac